frontend/app/views/spree/checkout/_delivery.html.erb

```diff
@@ -38,6 +38,7 @@
           <h5 class="stock-shipping-method-title"><%= Spree.t(:shipping_method) %></h5>
           <ul class="field radios shipping-methods">
             <% ship_form.object.shipping_rates.each do |rate| %>
+              <% next unless ship_form.object.selected_shipping_rate == rate || rate.shipping_method.available_to_users? %>
```

Comment:
I like this idea. I actually was thinking about it after I submitted this PR and thought that would clean it up adding a method like that. I'll go ahead and do that.